QSPIFBlockDevice.cpp

```diff
@@ -0,0 +1,1263 @@
+        QSPIF_LOG(QSPIF_DEBUG_ERROR, "ERROR: init - _is_mem_ready Failed");
+        status = QSPIF_BD_ERROR_READY_FAILED;
+        goto Exit_Point;
+    }
+
+    /**************************** Parse SFDP Header ***********************************/
+    if ( 0 != _sfdp_parse_sfdp_headers(basic_table_addr, basic_table_size, sector_map_table_addr, sector_map_table_size)) {
+        QSPIF_LOG(QSPIF_DEBUG_ERROR, "ERROR: init - Parse SFDP Headers Failed");
+        status = QSPIF_BD_ERROR_PARSING_FAILED;
+        goto Exit_Point;
+    }
+
+
+    /**************************** Parse Basic Parameters Table ***********************************/
+    if ( 0 != _sfdp_parse_basic_param_table(basic_table_addr, basic_table_size) ) {
+        QSPIF_LOG(QSPIF_DEBUG_ERROR, "ERROR: init - Parse Basic Param Table Failed");
+        status = QSPIF_BD_ERROR_PARSING_FAILED;
+        goto Exit_Point;
+    }
+
+    /**************************** Parse Sector Map Table ***********************************/
+    _region_size_bytes[0] =
+        _device_size_bytes; // If there's no region map, we have a single region sized the entire device size
+    _region_high_boundary[0] = _device_size_bytes - 1;
+
+    if ( (sector_map_table_addr != NULL) && (0 != sector_map_table_size) ) {
+        QSPIF_LOG(QSPIF_DEBUG_INFO, "INFO: init - Parsing Sector Map Table - addr: 0x%xh, Size: %d", sector_map_table_addr,
+                  sector_map_table_size);
+        if ( 0 != _sfdp_parse_sector_map_table(sector_map_table_addr, sector_map_table_size) ) {
+            QSPIF_LOG(QSPIF_DEBUG_ERROR, "ERROR: init - Parse Sector Map Table Failed");
+            status = QSPIF_BD_ERROR_PARSING_FAILED;
+            goto Exit_Point;
+        }
+    }
+
+    // Configure  BUS Mode to 1_1_1 for all commands other than Read
+    _qspi_configure_format(QSPI_CFG_BUS_SINGLE, QSPI_CFG_BUS_SINGLE, QSPI_CFG_ADDR_SIZE_24, QSPI_CFG_BUS_SINGLE,
+                           QSPI_CFG_ALT_SIZE_8, QSPI_CFG_BUS_SINGLE, 0);
+
+    _is_initialized = true;
+
+Exit_Point:
+    _mutex->unlock();
+
+    return status;
+}
+
+int QSPIFBlockDevice::deinit()
+{
+    _mutex->lock();
+    if (_is_initialized == false) {
+        _mutex->unlock();
+        return QSPIF_BD_ERROR_OK;
+    }
+
+    // Disable Device for Writing
+    qspi_status_t status = _qspi_send_general_command(QSPIF_WRDI, -1, NULL, 0, NULL, 0);
+    int result = QSPIF_BD_ERROR_OK;
+
+    if (status != QSPI_STATUS_OK)  {
+        QSPIF_LOG(QSPIF_DEBUG_ERROR, "ERROR: Write Disable failed");
+        result = QSPIF_BD_ERROR_DEVICE_ERROR;
+    }
+    _is_initialized = false;
+    _mutex->unlock();
+
+    return result;
+}
+
+
+int QSPIFBlockDevice::read(void *buffer, bd_addr_t addr, bd_size_t size)
+{
+
+
+    int status = QSPIF_BD_ERROR_OK;
+
+    QSPIF_LOG(QSPIF_DEBUG_INFO, "INFO Inst: 0x%xh", _read_instruction);
+
+    _mutex->lock();
+
+    // Configure Bus for Reading
+    _qspi_configure_format(
+        _inst_width, //Bus width for Instruction phase
```

Comment:
why arg is each one on the new line? line 295 does not follow this. I think should be on the same line, the comments do not add much if we name variables correctly (_address_width = bus width for address phase) so would remove them anyway

Reply:
fixed